packages/console/src/pages/WebhookDetails/WebhookSettings/CustomHeaderField/index.tsx

```diff
@@ -37,101 +33,81 @@ function CustomHeaderField() {
-      : true;
-  };
+      return getValues(`headers.${index}.key`)
+        ? Boolean(value) || t('webhook_details.settings.value_missing_error')
+        : true;
+    },
+    [getValues, t]
+  );
 
-  const revalidate = () => {
+  const revalidate = useCallback(() => {
     for (const [index] of fields.entries()) {
       void trigger(`headers.${index}.key`);
       if (submitCount > 0) {
         void trigger(`headers.${index}.value`);
       }
     }
-  };
+  }, [fields, submitCount, trigger]);
+
+  const getInputFieldProps = useMemo(
+    () => ({
+      key: (index: number) =>
+        register(`headers.${index}.key`, {
+          validate: (key) => keyValidator(key, index),
+          onChange: revalidate,
+        }),
+      value: (index: number) =>
+        register(`headers.${index}.value`, {
+          validate: (value) => valueValidator(value, index),
+          onChange: revalidate,
+        }),
+    }),
+    [keyValidator, register, revalidate, valueValidator]
+  );
 
   return (
     <FormField
       title="webhook_details.settings.custom_headers"
       tip={t('webhook_details.settings.custom_headers_tip')}
     >
-      {fields.map((header, index) => {
-        return (
-          <div key={header.id} className={styles.field}>
-            <div className={styles.input}>
-              <TextInput
-                className={styles.keyInput}
-                placeholder="Key"
-                error={Boolean(headerErrors?.[index]?.key)}
-                {...register(`headers.${index}.key`, {
-                  validate: (key) => keyValidator(key, index),
-                  onChange: revalidate,
-                })}
-              />
-              <TextInput
-                className={styles.valueInput}
-                placeholder="Value"
-                error={Boolean(headerErrors?.[index]?.value)}
-                {...register(`headers.${index}.value`, {
-                  validate: (value) => valueValidator(value, index),
-                  onChange: revalidate,
-                })}
-              />
-              {fields.length > 1 && (
-                <IconButton
-                  onClick={() => {
-                    remove(index);
-                  }}
-                >
-                  <Minus />
-                </IconButton>
-              )}
-            </div>
-            {headerErrors?.[index]?.key?.message && (
-              <div className={styles.error}>{headerErrors[index]?.key?.message}</div>
-            )}
-            {headerErrors?.[index]?.value?.message && (
-              <div className={styles.error}>{headerErrors[index]?.value?.message}</div>
-            )}
-          </div>
-        );
-      })}
-      <Button
-        size="small"
-        type="text"
-        title="general.add_another"
-        icon={<CirclePlus />}
-        onClick={() => {
-          append({ key: '', value: '' });
-        }}
+      <KeyValueInputField
```

Comment:
nit: How about we name it as `KeyValueFieldGroup`?

Reply:
Feels like Field/Group is similar. KeyValueInputField or KeyValueInputGroup both works for me.

Reply:
move foward